exporter/otlphttpexporter/otlp.go

```diff
@@ -106,7 +129,18 @@ func (e *baseExporter) pushMetrics(ctx context.Context, md pmetric.Metrics) erro
 
 func (e *baseExporter) pushLogs(ctx context.Context, ld plog.Logs) error {
 	tr := plogotlp.NewExportRequestFromLogs(ld)
-	request, err := tr.MarshalProto()
+
+	var err error
+	var request []byte
+	switch e.config.Encoding {
+	case EncodingJSON:
+		request, err = tr.MarshalJSON()
+	case EncodingProto:
+		request, err = tr.MarshalProto()
+	default:
```

Comment:
Since we have explicit cases for the encodings we handle I think this should error instead of fallback to proto.

Reply:
Since there is an issue with adding the default value explicitly (breaking contrib) I cant change this, unless breaking contrib is acceptable (I can create a small fix in the `f5cloudexporter` once this change is merged)